keyberon/src/layout.rs

```diff
@@ -1215,24 +1215,28 @@ impl<'a, const C: usize, const R: usize, const L: usize, T: 'a + Copy + std::fmt
             self.dequeue(queued);
         }
     }
-    fn press_as_action(&self, coord: KCoord, layer: usize) -> &'a Action<'a, T> {
+    fn resolve_coord(&self, coord: KCoord) -> &'a Action<'a, T> {
         use crate::action::Action::*;
-        let action = self
-            .layers
-            .get(layer)
-            .and_then(|l| l.get(coord.0 as usize))
-            .and_then(|l| l.get(coord.1 as usize));
-        match action {
-            None => &NoOp,
-            Some(Trans) => {
-                if layer != self.default_layer {
-                    self.press_as_action(coord, self.default_layer)
-                } else {
-                    &NoOp
+        for layer in self
+            .active_held_layers()
+            .iter()
+            .rev()
+            .chain(&[self.default_layer])
+        {
+            let action = self
+                .layers
+                .get(*layer)
+                .and_then(|l| l.get(coord.0 as usize))
```

Comment:
I think this would be better if the code first asserted that coord fits inside the layout(which it should) and then use the panicking indexing operator `[..]` - which should be able to have its bound checks removed by the compiler due to earlier asserts - hopefully.

The same applies to layer too.

Reply:
776f502